### PR TITLE
added patch logic for services

### PR DESCRIPTION
**Pull request by facchettos**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
part of eng-4141

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster ...

**What else do we need to know?**

### CODE REVIEW

File: pkg/controllers/resources/services/syncer.go [hunk 1]
```diff
@@ -71,45 +92,22 @@ func (s *serviceSyncer) Sync(ctx *synccontext.SyncContext, pObj client.Object, v
...
-		newService.Status = pService.Status
-		ctx.Log.Infof("update virtual service %s/%s, because status is out of sync", vService.Namespace, vService.Name)
-		translator.PrintChanges(vService, newService, ctx.Log)
-		err := ctx.VirtualClient.Status().Update(ctx, newService)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{Requeue: true}, err
```
Comment (FabianKramm) [inline pkg/controllers/resources/services/syncer.go:103]: This should go after line 98

File: pkg/controllers/resources/services/syncer.go [hunk 2]
```diff
@@ -63,53 +66,69 @@ func (s *serviceSyncer) Sync(ctx *synccontext.SyncContext, pObj client.Object, v
...
+		}
+		if retErr != nil {
+			s.EventRecorder().Eventf(vObj, "Warning", "SyncError", "Error syncing: %v", retErr)
+		}
+	}()
+
+	// check
+	sourceService, targetService := synccontext.SyncSourceTarget(ctx, pService, vService)
+
+	// check if we need to update
```
Comment (facchettos) [inline pkg/controllers/resources/services/syncer.go:102]: probably not up to date

File: pkg/controllers/resources/services/syncer.go [hunk 3]
```diff
@@ -63,53 +66,69 @@ func (s *serviceSyncer) Sync(ctx *synccontext.SyncContext, pObj client.Object, v
...
+	targetService.Spec.LoadBalancerIP = sourceService.Spec.LoadBalancerIP
+	targetService.Spec.Ports = sourceService.Spec.Ports
+	targetService.Spec.PublishNotReadyAddresses = sourceService.Spec.PublishNotReadyAddresses
+	targetService.Spec.Type = sourceService.Spec.Type
+	targetService.Spec.ExternalName = sourceService.Spec.ExternalName
+	targetService.Spec.ExternalTrafficPolicy = sourceService.Spec.ExternalTrafficPolicy
+	targetService.Spec.SessionAffinity = sourceService.Spec.SessionAffinity
+	targetService.Spec.SessionAffinityConfig = sourceService.Spec.SessionAffinityConfig
+	targetService.Spec.LoadBalancerSourceRanges = sourceService.Spec.LoadBalancerSourceRanges
+	targetService.Spec.HealthCheckNodePort = sourceService.Spec.HealthCheckNodePort
```
Comment (facchettos) [inline pkg/controllers/resources/services/syncer.go:113]: couldn't this be 
```go
targetService.Spec = sourceService.Spec 
```
and do what we need with the selectors beforehand?

Comment (FabianKramm) [reply pkg/controllers/resources/services/syncer.go:113]: There is many things that shouldn't be updated here (ClusterIPs, IPFamily etc), so its not a bad idea to have a whitelist here

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**